### ISSUE TITLE
[skill-request] spatium — Embed spatial proteomics single-cell protein expression profiles with a pretrain

### ISSUE THREAD

**Issue by matthewhirschey** [heurekalabsco/heurekaskills#9]
### What should the skill do?

Embed spatial proteomics single-cell protein expression profiles with a pretrained foundation model for cell-type annotation, spatial niche identification, and missing-marker imputation.

### Category

`models`

### Details

**Artifact:** Spatium
**Docs / code:** https://github.com/ploughhh/Spatium

Surfaced automatically by the *Agentic Discovery* newsletter's skill-discovery loop, which
reads each published issue plus the full text of every source and looks for callable
artifacts a skill could wrap.

**Why now:** bioRxiv preprint introducing a spatial-proteomics protein language foundation model with public code and pretrained weights.

**Evidence from the source** (verbatim):

> The code for Spatium, including model implementation, pretraining weights, and all downstream documentations, is publicly available at https://github.com/ploughhh/Spatium

The `docs_url` above was verified to appear in the fetched source and to resolve (not a
404). Confidence 0.7 against a 0.7 floor.

### Example source or paper (URL / DOI)

https://biorxiv.org/content/10.64898/2026.07.23.740264

### Known license of the source (if any)

unknown (code public; fine-tuning datasets to be released upon publication)

---

<sub>Filed by the newsletter discovery loop · origin `newsletter` · from
[issue 65](https://newsletter.heurekalabs.co/openai-agents-rewrite-scientific-code/) — "Protein language model for spatial proteomics" ·
dedupe `ff287fc95f52`</sub>

**Comment by matthewhirschey** [comment]
**Licence re-check — the licence questions resolve; the blocker is upstream maturity.**

Both objections in the earlier note are answerable, and the record should say so.

1. **The copyright holder is stated.** The `LICENSE` appendix is still unfilled boilerplate, but `docs/conf.py` in the same repository carries `copyright = '2026, Tiangang Wang'`. bioRxiv lists Wang, T. as first author of the preprint; the corresponding author is Xiaobo Zhou at the University of Texas Health Science Center at Houston, and the preprint itself is CC-BY.

2. **The weights are covered by the repository licence.** The pretrained checkpoint `final.ckpt` (~81 MB) is committed inside the repository, so it is part of the Work the Apache-2.0 `LICENSE` governs. There is no artifact sitting outside the licensed tree waiting on separate terms. No HuggingFace mirror exists; the in-repo file is the distribution.

The "credited to an individual rather than the publishing project" rule applies to content this registry would redistribute. A skill for this model would be written here and licensed CC-BY-4.0, carrying no upstream text, so that rule does not reach it — `skills/autodock-vina` is the existing shape: written here, and it routes users to GPL-2.0 engines.

What actually stops it is that the project is not yet runnable as documented:

- `docs/installation.rst` is empty; the hosted page renders as a heading with no body.
- `docs/tutorials.rst` reads "The documentation is still underconstruction, this part will be finish very soon."
- There is no `pyproject.toml` or `setup.py`, and no package on PyPI (`spatium` there is an unrelated spatial-math library). The one worked example imports as `from model import spaProFormer` — run from inside `src/`.
- Input must be Zarr in a schema produced only by the two tokenizer notebooks, and the fine-tuning datasets are still unreleased.
- No inference-only path is documented. The documented entry point is fine-tuning, which needs labels and a GPU — while the request is for embedding extraction.

This registry requires every code block to be run before it ships, so building this today means reverse-engineering `src/` and reproducing a fine-tune from the checkpoint against a dataset we assemble ourselves. That is a maturity problem upstream, not a licence problem here.

Staying open with `needs-info`. Re-check when installation instructions or the tutorials land, or when an installable package appears.

**Comment by matthewhirschey** [comment]
**Re-check 2026-09-07 — upstream moved, but the stated condition is not met. Still blocked, and now for a narrower reason.**

The August note said to re-check "when installation instructions or the tutorials land, or when
an installable package appears." Upstream has been active — pushes on 4 and 7 September,
including `Revise README with comprehensive project details` — and the README now has
`## Installation`, `## Quick Start`, `## Pretrained Models` and `## License` sections that did
not exist before. On a surface read this looks resolved. It is not, and the specific reasons
have changed enough to be worth writing down so the next pass tests the right things.

**The install line points at a file that does not exist.** The README instructs:

```bash
mamba create -n spatium python=3.9
pip install -r requirements.txt
```

There is no `requirements.txt` at the repository root. The full tree is 33 blobs and the only
file of that name is `docs/requirements.txt`, which is the Sphinx build's own dependency list
(it exists to serve `.readthedocs.yaml`, not the package). So the documented install cannot be
run as written. Still no `pyproject.toml` or `setup.py`, and the `spatium` on PyPI is still the
unrelated spatial-math library (v1.5.5, "a fast spatial math library"), not this project.

**Quick Start stops one line short of being a Quick Start.** The whole of "Load a pretrained
Spatium model" is:

```python
import torch
checkpoint = torch.load("final.ckpt", map_location="cpu")
```

That produces a state dict. Nothing in the README instantiates a model, loads the weights into
it, or computes an embedding — which is what this request is for. `import torch` is the only
import statement anywhere in the file.

**The docs still are not there.** `docs/installation.rst` is 2 lines; `docs/tutorials.rst` still
carries the under-construction sentence. The README itself says "Detailed environment
information will be provided with the public release," so upstream considers this pre-release.

Licence remains Apache-2.0 with `final.ckpt` committed in-tree, so the August licence analysis
still holds — that was never the blocker.

**Sharper re-check condition,** replacing the earlier one, which the README revision would have
technically satisfied without making the project runnable: a root `requirements.txt` (or a
`pyproject.toml`) that actually resolves, **and** a documented path from `final.ckpt` to an
embedding. Either alone is not enough.
